scripts/src/saforcharttesting/saforcharttesting.py

```diff
@@ -242,65 +241,44 @@ def delete_clusterrolebinding(name):
         sys.exit(1)
 
 
-def write_sa_token(namespace, token):
-    secret_found = False
-    secrets = []
+def write_sa_token(namespace, token_file):
+    """Write's the service account token to token_file."""
+    token_found = False
     for i in range(7):
+        # On retry, wait a little extra time before starting to give the cluster
+        # time to process the resources created before this.
+        if i > 0:
+            time.sleep(5)
+            print(f"[INFO] looking for service account token (retry {i})")
         out = subprocess.run(
-            ["oc", "get", "serviceaccount", namespace, "-n", namespace, "-o", "json"],
+            ["oc", "get", "secret", namespace, "-n", namespace, "-o", "json"],
             capture_output=True,
         )
         stdout = out.stdout.decode("utf-8")
         if out.returncode != 0:
             stderr = out.stderr.decode("utf-8")
             if stderr.strip():
-                print("[ERROR] retrieving ServiceAccount:", namespace, stderr)
-                time.sleep(10)
-        else:
-            sa = json.loads(stdout)
-            if len(sa["secrets"]) >= 2:
-                secrets = sa["secrets"]
-                secret_found = True
-                break
-            else:
-                pattern = r"Tokens:\s+([A-Za-z0-9-]+)"
-                dout = subprocess.run(
-                    ["oc", "describe", "serviceaccount", namespace, "-n", namespace],
-                    capture_output=True,
-                )
-                dstdout = dout.stdout.decode("utf-8")
-                match = re.search(pattern, dstdout)
-                if match:
-                    token_name = match.group(1)
-                else:
-                    print("[ERROR] Token not found, Exiting")
-                    sys.exit(1)
-                secrets.append({"name": token_name})
-                secret_found = True
-                break
-        time.sleep(10)
```

Comment:
Do you still need/wait to sleep some? Or is this no longer important/needed because the secrete is being proactively created?

Reply:
The sleep was moved to the top of the function, and is only executed on retries.